src/server/authentication.js

```diff
@@ -0,0 +1,111 @@
+const argon2 = require('argon2');
+const validator = require('validator');
+const jwt = require('jsonwebtoken');
+
+module.exports = psql => ({
+
+  /*
+      first checks if username/email already exist
+      returns true if successful
+     */
+  register: async (user) => {
+    const { username } = user;
+    const { email } = user;
+    const { password } = user;
+    let checkUsers;
+    let hash;
+
+    if (!validator.isEmail(email)) {
+      return false;
+    }
+
+    try {
+      checkUsers = await psql('users')
+        .where('username', username)
+        .orWhere('email', email)
+        .returning('user_id');
+
+      if (checkUsers.length > 0) {
+        return 'username or email already taken';
+      }
+
+      hash = await argon2.hash(password);
+      await psql('users').insert({
+        username,
+        email,
+        password: hash,
+        date_created: new Date(),
+      }).returning('username');
```

Comment:
Currently the length limit for the password column is 100 characters. Do you think that is too small for the hashing done here?

Reply:
Tried with several password lengths, the hashed passwords are of length 95 characters so this should be fine.